packages/trpc/server/routers/viewer/workflows/update.handler.ts

```diff
@@ -425,11 +424,21 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
 
       //step was edited
     } else if (JSON.stringify(oldStep) !== JSON.stringify(newStep)) {
-      if (!hasPaidPlan && !isSMSOrWhatsappAction(oldStep.action) && isSMSOrWhatsappAction(newStep.action)) {
-        throw new TRPCError({ code: "UNAUTHORIZED" });
+      // check if step that require team plan already existed before
+      if (
+        !hasPaidPlan &&
+        !isTextMessageToSpecificNumber(oldStep.action) &&
+        isTextMessageToSpecificNumber(newStep.action)
+      ) {
+        throw new TRPCError({ code: "UNAUTHORIZED", message: "Not available on free plan" });
       }
-      if (!isKYCVerified && isTextMessageToAttendeeAction(newStep.action)) {
-        throw new TRPCError({ code: "UNAUTHORIZED", message: "Account needs to be verified" });
+      // check if step that require team org already existed before
+      if (
+        !hasOrgsPlan &&
+        isTextMessageToAttendeeAction(newStep.action) &&
+        !isTextMessageToAttendeeAction(oldStep.action)
```

Comment:
Only throw an unauthorized error if the edited step was not a 'text message to attendee' action already before